lib/fs.js

```diff
@@ -2034,6 +2034,11 @@ ReadStream.prototype.close = function(cb) {
   this.fd = null;
 };
 
+// needed because as it will be called with arguments
+// that does not match this.close() signature
+function closeOnOpen() {
+  this.close();
```

Comment:
yes, that's the exact meaning. `this`  will be the stream object. I'll add a comment to note that.